src/main/java/com/synopsys/integration/detect/lifecycle/run/DetectRun.java

```diff
@@ -82,19 +79,17 @@ public void run(BootSingletons bootSingletons) {
             if (productRunData.shouldUseBlackDuckProduct()) {
                 BlackDuckRunData blackDuckRunData = productRunData.getBlackDuckRunData();
                 if (blackDuckRunData.isNonPersistent() && blackDuckRunData.isOnline()) {
-                    RapidModeStepRunner rapidModeSteps = new RapidModeStepRunner(operationRunner, stepHelper, bootSingletons.getGson(), bootSingletons.getDirectoryManager());
-                    
-                    Optional<String> scaaasFilePath = bootSingletons.getDetectConfigurationFactory().getScaaasFilePath();
-                    
-                    rapidModeSteps.runOnline(blackDuckRunData, nameVersion, bdio, universalToolsResult.getDockerTargetData(), scaaasFilePath);
+                    RapidModeStepRunner rapidModeSteps = new RapidModeStepRunner(operationRunner, stepHelper, bootSingletons.getGson(), bootSingletons.getDirectoryManager(), bootSingletons.getDetectConfigurationFactory());
+
```

Comment:
Tried to follow the design patters [here](https://sig-confluence.internal.synopsys.com/display/integration/Design+priorities), particularly what's said under naming and dependency injection ... but there is more than one way to interpret those notes ..